lib/pavilion/scheduler_plugins.py

```diff
@@ -1,39 +1,71 @@
 import collections
 from pavilion.variables import VariableSetManager
-from yapsy.IPlugin import IPlugin
+from yapsy import IPlugin
 import logging
 import re
 
 LOGGER = logging.getLogger('pav.{}'.format(__name__))
 
-class PluginSchedulerError(RuntimeError):
+class SchedulerPluginError(RuntimeError):
     pass
 
-_SCHEDULER_PLUGINS = {}
+_SCHEDULER_PLUGINS = None
+_LOADED_PLUGINS = None
+
+_SCHEDULER_VARS = [ 'num_nodes', 'procs_per_node', 'qos', 'reservation',
+                    'partition', 'account', 'down_nodes', 'unused_nodes',
+                    'busy_nodes', 'maint_nodes', 'other_nodes', 'chunk_size' ]
```

Comment:
I'm pretty sure the node counts are going to have to be per partition, which means partition should be a complex variable with those as attributes.